lib/spack/spack/version/version_types.py

```diff
@@ -135,79 +194,79 @@ class StandardVersion(ConcreteVersion):
 
     __slots__ = ["version", "string", "separators"]
 
-    def __init__(self, string: Optional[str], version: Tuple[tuple, tuple], separators: tuple):
+    def __init__(self, string: str, version: Tuple[tuple, tuple], separators: tuple):
         self.string = string
         self.version = version
         self.separators = separators
 
     @staticmethod
-    def from_string(string: str):
+    def from_string(string: str) -> "StandardVersion":
         return StandardVersion(string, *parse_string_components(string))
 
     @staticmethod
-    def typemin():
+    def typemin() -> "StandardVersion":
         return _STANDARD_VERSION_TYPEMIN
 
     @staticmethod
     def typemax():
         return _STANDARD_VERSION_TYPEMAX
 
-    def __bool__(self):
+    def __bool__(self) -> bool:
         return True
 
-    def __eq__(self, other):
+    def __eq__(self, other) -> bool:
```

Comment:
As above, I think the usual signature is:
```suggestion
    def __eq__(self, other: object) -> bool:
```

Reply:
fixed all but the exception above.